### PR TITLE
add infof() logging utility

### PR DESCRIPTION
**Pull request by gfx**
This is a debug feature to automatically add a timestamp to logging output.

### CODE REVIEW

**Comment by syohex** [comment]
https://en.cppreference.com/w/c/numeric/math/log

I suppose it should be renamed because `logf` is declared in math.h C standard library. (e.g. `h2o_log`, `h2olog_log` etc).

**Comment by gfx** [comment]
OMG. Changed it to `infof()`.

File: h2olog.cc [hunk 1]
```diff
@@ -44,24 +45,44 @@ Other options:
...
+    localtime_r(&t, &tms);
+    const char *iso8601format = "%FT%TZ";
+    strftime(buf, buf_len, iso8601format, &tms);
+}
+
+static void infof(const char* fmt, ...)
+    __attribute__((format (printf, 1, 2)));
+
+static void infof(const char* fmt, ...)
+ {
```
Comment (syohex) [inline h2olog.cc:61]: Indentation is strange. Do `clang-format -i h2olog.cc`

Comment (gfx) [reply h2olog.cc]: Applied. Thanks 😭

**Pull request merged**